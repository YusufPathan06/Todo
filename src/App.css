.App {
  max-width: 600px;
  margin: 0 auto;
  padding: 2rem;
  background-color: #f5f5f5;
  border-radius: 8px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  font-family: sans-serif;
}

h1 {
  text-align: center;
  margin-bottom: 2rem;
  color: #333;
}

input[type='text'] {
  padding: 0.5rem;
  border: none;
  border-radius: 4px;
  box-shadow: inset 0 0 4px rgba(0, 0, 0, 0.1);
  font-size: 1.2rem;
}

button {
  padding: 0.5rem 1rem;
  background-color: #0077cc;
  color: #fff;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
}

button:hover {
  background-color: #0066b3;
}

ul {
  list-style: none;
  padding: 0;
  margin: 2rem 0;
}

li {
  display: flex;
  align-items: center;
  margin: 1rem 0;
}

li input[type='text'] {
  margin-right: 1rem;
  flex-grow: 1;
  color: #555;
}

li button {
  background-color: #cc0000;
  margin-left: 1rem;
}

li button:hover {
  background-color: #b30000;
}
