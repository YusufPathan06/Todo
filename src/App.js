import React, { useState } from 'react';
import './App.css';

function App() {
  const [todos, setTodos] = useState([]);
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleAddTodo = () => {
    if (inputValue.trim()) {
      setTodos([...todos, { text: inputValue.trim() }]);
      setInputValue('');
    }
  };

  const handleEditTodo = (index, newText) => {
    const updatedTodos = [...todos];
    updatedTodos[index] = { text: newText };
    setTodos(updatedTodos);
  };

  const handleDeleteTodo = (index) => {
    const updatedTodos = [...todos];
    updatedTodos.splice(index, 1);
    setTodos(updatedTodos);
  };

  return (
    <div className="App">
      <h1>My Todo List</h1>
      <div className="input-container">
        <input
          type="text"
          placeholder="Add a todo"
          value={inputValue}
          onChange={handleInputChange}
        />
        <button onClick={handleAddTodo}>Add</button>
      </div>
      <ul>
        {todos.map((todo, index) => (
          <li key={index}>
            <input
              type="text"
              value={todo.text}
              onChange={(e) => handleEditTodo(index, e.target.value)}
            />
            <button onClick={() => handleDeleteTodo(index)}>Delete</button>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default App;
