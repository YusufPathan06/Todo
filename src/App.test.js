import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";

test("todo component is rendered", () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});

test("add, edit and delete todo", () => {
  render(<App />);

  const addTodo = screen.getByTestId("add-todo");
  fireEvent.change(addTodo, { target: { value: "yusuf" } });

  const addButton = screen.getByTestId("add-button");
  fireEvent.click(addButton);

  const editTodo = screen.getByTestId("edit-todo");
  fireEvent.change(editTodo, { target: { value: "Yusuf Pathan" } });

  const deleteButton = screen.getByTestId("delete-button");
  fireEvent.click(deleteButton);

  fireEvent.change(addTodo, { target: { value: "" } });
  fireEvent.click(addButton);
})
