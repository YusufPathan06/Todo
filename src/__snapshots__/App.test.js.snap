// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo component is rendered 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      My Todo List
    </h1>
    <div
      class="input-container"
    >
      <input
        data-testid="add-todo"
        placeholder="Add a todo"
        type="text"
        value=""
      />
      <button
        data-testid="add-button"
      >
        Add
      </button>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
